### PR TITLE
WIP - POC to limit kubelet to modifying well-known labels in NodeRestriction admission

### PR DESCRIPTION
**Pull request by liggitt**
This is a proof of concept related to https://github.com/kubernetes/community/pull/911/files#r188809456

It moves any label changes from a kubelet outside a well-known set of labels to an annotation

That does several things:
* preserves the label information the kubelet was trying to set
* makes that information observable via a central controller with higher permissions that could promote those to be real labels on the Node object if desired
* avoids giving kubelets direct control over steering workloads to themselves
* avoids disrupting normal registration/heartbeating activities of kubelets
* avoids needing to configure the NodeRestriction admission plugin

Other experimental changes:
* remove dependence on node self-labeling for fluentd daemonset (supposed to be removed in 1.8)
* remove dependence on node self-labeling for metadata proxy (unclear why this was done, will track down the owner)
* identify remaining self-labeling uses... in-tree, appears to just be the nvidia device plugin...

### CODE REVIEW

File: plugin/pkg/admission/noderestriction/admission.go [hunk 1]
```diff
@@ -344,6 +368,9 @@ func (c *nodePlugin) admitNode(nodeName string, a admission.Attributes) error {
 		if !apiequality.Semantic.DeepEqual(node.Spec.Taints, oldNode.Spec.Taints) {
 			return admission.NewForbidden(a, fmt.Errorf("cannot modify taints"))
 		}
+
+		// On update, filter and record any disallowed labels
+		filterLabels(node, oldNode.Labels)
```
Comment (k82cn) [inline plugin/pkg/admission/noderestriction/admission.go]: why not reject it directly? will annotation be large?

**Comment by liggitt** [comment]
interestingly, e2e failed with these recorded:

> `kubernetes.io/node-restriction.disallowed-labels: '{"beta.kubernetes.io/fluentd-ds-ready":"true","beta.kubernetes.io/metadata-proxy-ready":"true"}'`

it appears the nodes are configured to label themselves as ready for particular workloads, which seems pretty odd

**Comment by mikedanese** [comment]
The labels that were removed for GCE in this PR can be readded by the gcp-controller-manager once we run it in GCE. I can create a PR.

**Pull request closed without merging**